libsplinter/src/network/dispatch.rs

```diff
@@ -289,6 +289,8 @@ pub trait MessageSender<R>: Send {
     ///
     /// If an error occurs, return the intended recipient and message bytes.
     fn send(&self, reciptient: R, message: Vec<u8>) -> Result<(), (R, Vec<u8>)>;
+
+    fn box_clone(&self) -> Box<dyn MessageSender<R>>;
```

Comment:
Why is this necessary?

Reply:
I added it for passing it into the create_authorization_dispatcher and thought it would be useful but I will rework that and remove it.

Reply:
auth2::handlers::create_authorization_dispatcher takes a MessageSender<ConnectionId>> and that needs to be wrapped in Box. That function clones the message sender to create two dispatchers.

Reply:
I was able to rework that function. removed box_clone